router/router.go

```diff
@@ -0,0 +1,188 @@
+package router
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/go-etcd/etcd"
+
+	"github.com/spesnova/iruka/schema"
+)
+
+const (
+	DefaultKeyPrefix = "/vulcand"
+	DefaultMachines  = "http://localhost:4001"
+)
+
+type Router struct {
+	etcd      etcd.Client
+	keyPrefix string
+}
+
+func NewRouter(machines, keyPrefix string) *Router {
+	if os.Getenv("IRUKA_ETCD_MACHINES") != "" {
+		machines = os.Getenv("IRUKA_ETCD_MACHINES")
+	}
+
+	if machines == "" {
+		machines = DefaultMachines
+	}
+
+	m := strings.Split(machines, ",")
+	etcdClient := *etcd.NewClient(m)
+	return &Router{etcdClient, keyPrefix}
+}
+
+// TODO(dtan4):
+//   Divide by upstream (blue/green)
+func (r *Router) AddBackend(name string) error {
+	backend := schema.VulcandBackend{
+		Type: "http",
+	}
+	j, err := marshal(backend)
+
+	if err != nil {
+		return err
+	}
+
+	key := path.Join(r.keyPrefix, "backends", name, "backend")
+	_, err = r.etcd.Create(key, string(j), 0)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Router) RemoveBackend(name string) error {
+	key := path.Join(r.keyPrefix, "backends", name, "backend")
+	_, err := r.etcd.Delete(key, true)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO(dtan4):
+//   Divide by upstream (blue/green)
+func (r *Router) AddServer(appID, containerName, url string) error {
+	server := schema.VulcandServer{
+		URL: url,
+	}
+	j, err := marshal(server)
+
+	if err != nil {
+		return err
+	}
+
+	key := path.Join(r.keyPrefix, "backends", appID, "servers", containerName)
+	_, err = r.etcd.Create(key, string(j), 0)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Router) RemoveServer(appID, containerName string) error {
+	key := path.Join(r.keyPrefix, "backends", appID, "servers", containerName)
+	_, err := r.etcd.Delete(key, true)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Router) IsServerExists(appID, containerName string) bool {
```

Comment:
Same as `AddServer()` and `RemoveServer()`, `IsServerExists` can take `schema.Container` instead of `appID` and `containerName`.